nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/test/groovy/org/apache/nifi/properties/NiFiPropertiesLoaderGroovyTest.groovy

```diff
@@ -472,4 +472,21 @@ class NiFiPropertiesLoaderGroovyTest extends GroovyTestCase {
 
         assert readPropertiesAndValues == readPasswordPropertiesAndValues
     }
+    
+    @Test
+    void testShouldStripWhitespace() throws Exception {
+        // Arrange
+        File unprotectedFile = new File("src/test/resources/conf/nifi_with_whitespace.properties")
+        NiFiPropertiesLoader niFiPropertiesLoader = new NiFiPropertiesLoader()
+
+        // Act
+        NiFiProperties niFiProperties = niFiPropertiesLoader.load(unprotectedFile.path)
+
+        // Assert
+        assert niFiProperties.getProperty("nifi.whitespace.propWithLeadingSpace") == "foo";
+        assert niFiProperties.getProperty("nifi.whitespace.propWithTrailingSpace") == "foo";
+        assert niFiProperties.getProperty("nifi.whitespace.propWithLeadingAndTrailingSpace") == "foo";
+        assert niFiProperties.getProperty("nifi.whitespace.propWithTrailingTab") == "foo";
```

Comment:
Looks like this test is failing in all the checks, looking at your regex, not sure you're currently removing tab characters

Reply:
I think I misunderstood how multiline properties worked and this change fixes that. That simplified the regex a lot. Users can still embed a newline character with "\n". I figure if they go to that much trouble, they should be allowed to. The tab issue has been fixed. I also moved the test into a test class that executes under Windows, where I'm working.

Reply:
JDK 11 build failed still, though can't see from the errors whether that's caused by your changes or an intermittent issue with the build (and therefore whether a simple re-execution of that check may let it pass)?